Sources/HummingbirdCore/Request/RequestBody+inboundClose.swift

```diff
@@ -0,0 +1,142 @@
+    ///
+    /// - Parameters
+    ///   - operation: The actual operation
+    ///   = onInboundClose: handler invoked when inbound is closed
+    /// - Returns: Return value of operation
+    @available(macOS 15, iOS 18, tvOS 18, *)
+    public func consumeWithInboundCloseHandler<Value: Sendable>(
+        _ operation: sending (RequestBody) async throws -> Value,
+        onInboundClosed: @Sendable @escaping () -> Void
+    ) async throws -> Value {
+        let iterator: UnsafeTransfer<NIOAsyncChannelInboundStream<HTTPRequestPart>.AsyncIterator> =
+            switch self._backing {
+            case .nioAsyncChannelRequestBody(let iterator):
+                iterator.underlyingIterator
+            default:
+                preconditionFailure("Cannot run consumeWithInboundCloseHandler on edited request body")
+            }
+        let (requestBody, source) = RequestBody.makeStream()
+        return try await withInboundCloseHandler(
+            iterator: iterator.wrappedValue,
+            source: source,
+            operation: {
+                try await operation(requestBody)
+            },
+            onInboundClosed: onInboundClosed
+        )
+    }
+
+    /// Run provided closure but cancel it if the inbound request part stream is closed.
+    ///
+    /// This function is designed for use with long running requests like server sent events. It assumes you
+    /// are not going to be using the request body after calling as it consumes the request body, it also assumes
+    /// you havent edited the request body prior to calling this function.
+    ///
+    /// If the response finishes the connection will be closed.
+    ///
+    /// - Parameter operation: The actual operation to run
+    /// - Returns: Return value of operation
+    public func consumeWithCancelOnInboundClose<Value: Sendable>(
+        _ operation: sending @escaping (RequestBody) async throws -> Value
+    ) async throws -> Value {
+        let (barrier, source) = AsyncStream<Void>.makeStream()
+        return try await consumeWithInboundCloseHandler { body in
+            let unsafeOperation = UnsafeTransfer(operation)
+            return try await withThrowingTaskGroup(of: Value.self) { group in
+                group.addTask {
+                    var iterator = barrier.makeAsyncIterator()
+                    _ = await iterator.next()
+                    throw CancellationError()
+                }
+                group.addTask {
+                    try await unsafeOperation.wrappedValue(body)
+                }
+                if case .some(let value) = try await group.next() {
+                    source.finish()
+                    return value
+                }
+                group.cancelAll()
+                throw CancellationError()
+            }
+        } onInboundClosed: {
+            source.finish()
+        }
+    }
+
+    @available(macOS 15, iOS 18, tvOS 18, *)
+    func withInboundCloseHandler<Value: Sendable>(
```

Comment:
```suggestion
    fileprivate func withInboundCloseHandler<Value: Sendable>(
```

Let's explicitly mark this as something for local use only